radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx

```diff
@@ -33,19 +34,43 @@ const renderTFValue = (val: string): JSX.Element => {
       </a>
     )
   }
+
   return (
     <div key={val} className="mb-2">
       {val}
     </div>
   )
 }
 
-const renderTFValues = (val: any): JSX.Element => {
-  if (Array.isArray(val)) {
-    return <>{val.map(renderTFValue)}</>
+const renderTFValues = (val: TF_OUTPUT_VARIABLE | null): JSX.Element => {
+  if (val === null || val.value === null) {
+    return <></>
+  }
+  if (typeof val.value === "object" && !Array.isArray(val.value)) {
+    return (
+      <>
+        {Object.entries(val.value).map(([objKey, objValue]) => (
+          <div key={objKey}>
+            {typeof objValue === "string" ? (
+              <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3">
+                <div className="text-left text-faint col-span-1">{objKey}</div>
+                <div className="col-span-2 sm:col-span-1 break-words">
+                  {renderTFValue(objValue)}
+                </div>
+              </div>
+            ) : (
+              <></>
+            )}
```

Comment:
Shouldn't we just call `renderTFValues` again (recursively)? This should remove duplicated code for rendering strings as well as handle nested objects... Thoughts?

Reply:
if not string then only we can call again and updated like that